test/executor.test.js

```diff
@@ -271,6 +271,67 @@ describe('executor', function() {
         done();
       });
     });
+
+    describe ('for mixins', function() {
+      it('defines mixins from instructions', function() {
+        var example = appdir.writeFileSync('mixins/example.js',
+          'module.exports = ' +
+          'function(Model, options) {}');
+
+        var ts = appdir.writeFileSync('mixins/time-stamps.js',
+          'module.exports = ' +
+          'function(Model, options) {}');
+
+        appdir.writeConfigFileSync('mixins/time-stamps.json', {
+          name: 'Timestamping'
+        });
+
+        var options = { appRootDir: appdir.PATH,
+          mixinDirs: ['mixins'] };
+        boot(app, options);
+
+        var modelBuilder = (app.registry || app.loopback).modelBuilder;
+
+        var registry = modelBuilder.mixins.mixins;
+        expect(Object.keys(registry)).to.eql(['Example', 'Timestamping']);
+
+        var meta = modelBuilder.mixins.meta;
+        var sourceFiles = meta.map(getSourceFileProperty);
+        expect(sourceFiles).to.eql([example, ts]);
+      });
+
+      it('applies mixin metadata specified in instructions', function() {
+        appdir.writeFileSync('mixins/time-stamps.js',
+          'module.exports = ' +
+          'function(Model, options) {}');
+
+        var json = appdir.writeConfigFileSync('mixins/time-stamps.json', {
+          name: 'Timestamping',
+          description: 'Track create and update',
+          options: {
+            postfix: {
+              'type': 'string',
+              'description': 'The postfix string',
+              'default': 'At'
+            }
+          }
+        });
+
+        var options = { appRootDir: appdir.PATH,
+          mixinDirs: ['mixins'] };
+        boot(app, options);
+
+        var modelBuilder = (app.registry || app.loopback).modelBuilder;
```

Comment:
I think it's ok to rely on `app.registry` being present when inside a unit test. loopback-boot controls the version of loopback used for unit-tests via the devDependency in `package.json`.

Could you please do the following?

 - bump up the [loopback version in package.json](https://github.com/strongloop/loopback-boot/blob/e3763ac28e723d351a1423abc8e068916fbe49d7/package.json#L41) to "^2.16.3" 
 - use `app.registry.modelBuilder` in unit-tests (`test/*.js`)

Note that `lib/*.js` files must support older loopback versions, so they should not be changed.